app/models/structural_metadata.rb

```diff
@@ -14,9 +14,21 @@
 
 class StructuralMetadata < ActiveFedora::Datastream
```

Comment:
Maybe a lot of this would be handled if StructuralMetadata subclassed NokogiriDatastream instead of just plain Datastream.